app/models/reference_token.rb

```diff
@@ -0,0 +1,3 @@
+class ReferenceToken < ApplicationRecord
+  belongs_to :reference, class_name: 'ApplicationReference'
```

Comment:
```suggestion
  belongs_to :application_reference
```

I think the reason we changed from using `reference` for the name is because it can conflict with build in methods in Rails